test/libsolidity/syntaxTests/inlineAssembly/shadowing/local_function_by_opcode.sol

```diff
@@ -0,0 +1,10 @@
+contract C {
+    function add(uint, uint) public pure {}
+    function g() public pure {
+        assembly {
+            let x := add(1, 2)
+            mstore(0, x)
+        }
+    }
+}
+// ----
```

Comment:
Could you please add a semantic test for the following if something like this does not already exist?

```
contract C {
    uint x;
    function add(uint, uint) public pure returns (uint) { return 0; }
    function g() public returns (uint) {
        x = add(1, 2);
        assembly {
            x := add(x, add(1, 2))
        }
    }
}
```

just to ensure that the intended add is called in each context.